### ISSUE TITLE
unhandled exception when refreshing access token

### ISSUE THREAD

**Issue by joeferraro** [kidtsunami/fetch-salesforce#15]
haven't dug into it yet, but figured i'd post here for now

<img width="374" alt="cursor_and_glass" src="https://cloud.githubusercontent.com/assets/54157/20455526/fd2ee7f2-ae2b-11e6-956b-da9329a7768d.png">

**Comment by joeferraro** [comment]
2 things here:

1. it doesn't seem like we need to bind(this) [here](https://github.com/kidtsunami/fetch-salesforce/blob/master/src/lib/fetcher.ts#L138)
2. one that bind is removed, we move forward with attempting to refresh our token, but our `tokenServiceURL` is null. Setting this explicitly in options resolves it. We should probably be deriving it from our instanceURL unless explicitly set in options

**Comment by kidtsunami** [comment]
This is resolved now.

Implemented over a variety of commits. 

We have removed the bind, as well as wrapping the returned fetch promise (with node-fetch we were able to set fetch.Promise to be bluebird's) with bluebird's so we are still using Bluebird's richer promises. This also involved switching to use isomorphic-fetch (which uses node-fetch) to handle if global.fetch is not defined. This is in b9bff6da9d69be9fca367fb7ca5f1a99d0f5447b

We are also now dynamically generating service urls 0ba6625a03e71470c0a853af9a71449cf2c9c1b6

```
yarn upgrade fetch-salesforce@0.0.13
```
